src/metabase/upload.clj

```diff
@@ -209,53 +195,76 @@
     "unnamed_column"
     (u/slugify (str/trim raw-name))))
 
-(defn- is-pk?
-  [[col-name type]]
-  (and (#{"id" "pk" "uuid" "guid"} (u/lower-case-en (name col-name)))
-       (pk-base-types type)))
-
-(defn- ->ordered-maps-with-pk-column
-  "Sets appropriate type information on the first PK column it finds, otherwise adds a new one.
-
-  Returns a *map* with two keys: `:extant-columns` and `:generated-columns`."
-  [name-type-pairs]
-  (if-let [[pk-name pk-base-type] (first (filter is-pk? name-type-pairs))]
-    {:extant-columns    (assoc (ordered-map/ordered-map name-type-pairs) pk-name (base-type->pk-type pk-base-type))
-     :generated-columns (ordered-map/ordered-map)}
-    {:extant-columns    (ordered-map/ordered-map name-type-pairs)
-     :generated-columns (ordered-map/ordered-map :id ::auto-incrementing-int-pk)}))
+(def auto-pk-column-name
+  "The name of the auto-incrementing PK column."
+  "_mb_row_id")
+
+(defn- remove-indices
+  "Removes the elements at the given indices from the collection. Indices is a set."
+  [indices coll]
+  (keep-indexed (fn [idx item]
+                  (when-not (contains? indices idx)
+                    item))
+                coll))
+
+(defn- indices-where
+  "Returns a lazy seq of the indices where the predicate is true."
+  [pred coll]
+  (keep-indexed (fn [idx item]
+                  (when (pred item)
+                    idx))
+                coll))
+
+(defn- parse-rows*
+  "Returns a lazy seq of parsed rows, given a sequence of upload types for each column.
+  Replaces empty strings with nil."
+  [col-upload-types rows]
+  (let [settings (upload-parsing/get-settings)
+        parsers  (map #(upload-parsing/upload-type->parser % settings) col-upload-types)]
+    (for [row rows]
+      (for [[value parser] (map-with-nils vector row parsers)]
+        (when (not (str/blank? value))
+          (parser value))))))
 
 (defn- rows->schema
-  "rows should be a lazy-seq"
+  "Rows should be a lazy-seq. This function hides the logic for ignoring any columns in the CSV that have the same
+  normalized name as the auto-pk-column-name."
   [header rows]
   (let [normalized-header (->> header
-                               (map normalize-column-name)
-                               (mbql.u/uniquify-names)
+                               (map normalize-column-name))
+        ;; remove columns from the rows with the same normalized name as the auto-pk-column-name
+        auto-pk-col-indices (set (indices-where #(= auto-pk-column-name %) normalized-header))
+        ;; remove the auto-pk column from each row as we parse all the rows
+        remove-auto-pk-cols (fn [row]
+                              (remove-indices auto-pk-col-indices row))
+        rows              (cond->> rows
+                            auto-pk-col-indices
```

Comment:
```suggestion
                            (seq auto-pk-col-indices)
```

empty set is considered truthy under cond

Reply:
Fixed